summary.Rmd

```diff
@@ -0,0 +1,40 @@
+Summary
+=======
+
+1. Use available tools
+    a. Use ready-made images (e.g. from Docker Hub)
+    b. Dockerfile generators (e.g. repo2docker)
+    c. Project templates (e.g. [cookie cutter](https://github.com/cookiecutter/cookiecutter))
+    d. Linters
+2. Use versioned images
+3. Format for clarity
+    a. Use indentation, line breaks and unabbreviated arguments
+    b. Clarity > brevity
+    c. Group related commands
+4. Document within the Dockerfile
+    a. Add comments explaining the steps
+    b. Use labels for metadata
+    c. Include build and run commands
+6. Specify software versions
+    a. Pin system library versions
+    b. Explictly list package versions to install
+    c. Comment why each is needed
+9. Use version control
+    a. Use Continuous Integration and automated testing
+    b. ~~Use one Dockerfile per project~~
+    c. ~~Use project templates (e.g. cookie cutter)~~
+7. Mount user data
+    a. Use bind mount for accessibility
+    b. Keep large or sensitive datasets out with `.dockerignore`
+    c. Document expected mounts for external resources
+5. Order the instructions
+    a. Least --> most changable for caching and fast builds
+    b. Regularly use `--no-cache` to catch problems
+8. Make it one-click runnable
+    a. Set default behaviour with `CMD` and `ENTRYPOINT`
```

Comment:
b. Add and document behaviour for an interactive usage for your own daily routine

Reply:
Ok, I've added "b. Document how to use it interactively".